integration/service/network_test.go

```diff
@@ -115,3 +116,35 @@ func TestDockerNetworkReConnect(t *testing.T) {
 	assert.NilError(t, err)
 	assert.Check(t, is.DeepEqual(n1, n2))
 }
+
+func TestDockerRestartWithAttachbleNetwork(t *testing.T) {
+	skip.If(t, testEnv.DaemonInfo.OSType == "windows")
+	defer setupTest(t)()
+	d := swarm.NewSwarm(t, testEnv)
+	defer d.Stop(t)
+	client := d.NewClientT(t)
+	defer client.Close()
+	ctx := context.Background()
+
+	name := t.Name() + "dummyNet"
+	net.CreateNoError(ctx, t, client, name,
+		net.WithDriver("overlay"),
+		net.WithAttachable(),
+	)
+
+	c1 := container.Create(ctx, t, client, func(c *container.TestContainerConfig) {
+		c.NetworkingConfig = &network.NetworkingConfig{
+			EndpointsConfig: map[string]*network.EndpointSettings{
+				name: {},
+			},
+		}
+	})
+
+	err := client.ContainerStart(ctx, c1, types.ContainerStartOptions{})
+	assert.NilError(t, err)
+
+	var timeout time.Duration = 20 * time.Second
+	err = client.ContainerRestart(ctx, c1, &timeout)
```

Comment:
does this work with a smaller restart value , default in compose and cli is `10s`

Reply:
@arkodg Stopping busybox container need at least 10 seconds since busybox does not have signal handler for SIGTERM. So I feel the timeout here needs to be longer than 10 seconds.

Reply:
ah how about 11 :trollface:

Reply:
I'm afraid setting timeout to 11s will cause CI flaky.